src/log.cpp

```diff
@@ -279,7 +279,7 @@ std::optional<bool> log_dir_writable()
 
 std::string get_log_file_path()
 {
-	return output_file_path_+lg::log_file_suffix;
+	return output_file_path_.empty() ? "" : output_file_path_+lg::log_file_suffix;
```

Comment:
IMO this should be made to return an `optional<string>`.

Reply:
I haven't used `optional` before but is this what you meant?

Reply:
Uh, looks like the change broke a lot of checks.

Reply:
I think it's fine to leave it as it is for now. Making it an optional just means everywhere it's used you need to either get the value assuming its value is present or provide some other default value or functionality if it's not.

Reply:
'Leave it as it is' - meaning before the change to optional?

Reply:
Yeah. Using optional would be the more technically correct way to do it, but I don't think it really matters much either way here.